tests/conftest.py

```
@@ -2,27 +2,38 @@
     mock_settings_module = MagicMock()
     mock_settings_module.settings = mock_settings_instance
 
     # Replace the actual config.settings module in sys.modules
-    # This needs to happen before any test imports config.settings
-    monkeypatch.setitem(sys.modules, "config.settings", mock_settings_module)
+    sys.modules["config.settings"] = mock_settings_module
 
-    yield
+    # Patch sys.exit to prevent it from terminating the test run
+    monkeypatch = MagicMock() # Create a mock monkeypatch for pytest_configure
+    monkeypatch.setattr(sys, "exit", lambda x: None)
```

Ruff (0.14.1)

Unused lambda argument: `x`

(ARG005)